include/Sample.h

```diff
@@ -25,44 +25,55 @@
 #ifndef LMMS_SAMPLE_H
 #define LMMS_SAMPLE_H
 
+#include <cmath>
 #include <memory>
-#include <samplerate.h>
 
+#include "AudioResampler.h"
 #include "Note.h"
 #include "SampleBuffer.h"
-#include "lmms_basics.h"
 #include "lmms_export.h"
 
 namespace lmms {
 class LMMS_EXPORT Sample
 {
 public:
+	// values for buffer margins, used for various libsamplerate interpolation modes
+	// the array positions correspond to the converter_type parameter values in libsamplerate
+	// if there appears problems with playback on some interpolation mode, then the value for that mode
+	// may need to be higher - conversely, to optimize, some may work with lower values
+	static constexpr auto s_interpolationMargins = std::array<int, 5>{64, 64, 64, 4, 4};
```

Comment:
Btw will it make sense to push this back to gigplayer ?

Reply:
Maybe, but if we remove libsamplerate and use our own interpolation functions, there's no need.